backport: bitcoin#16333, #21862, #22385, #22550, #22597, #22632, #22718, #22907 - fire up test chains by first block - 2/n

knst opened the pull request:
## Issue being fixed or feature implemented
Backports from bitcoin related to hard-fork mechanism and accelerated action

## What was done?
see commits for backports.
Also fixed an issue of using from ChainState on RegTest: should be used specified chain, not the "best chain".


## How Has This Been Tested?
Run unit & functional tests

## Breaking Changes
That's a breaking changes for RegTest

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

UdjinM6 reviewed (dismissed):
utACK 90e84fd2c8808461a5a19e8676e3bccd2612078d

PastaPastaPasta commented on doc/release-notes-22632.md:
```diff
@@ -0,0 +1,8 @@
+Tests
+-----
+
+- For the `regtest` network the activation heights of several softforks were
+  changed.
+  * BIP 34 (blockheight in coinbase) from 500 to 2 (#16333)
+  * BIP 66 (DERSIG) from 1251 to 102 (#22632)
+  * BIP 65 (CLTV) from 1351 to 111 (#21862)
```
let's fix it here

PastaPastaPasta reviewed (approved):
utACK e4e7c440f48058dabdc571a0a20626b513efd26c

knst commented:
Range-diff:
```bash
$ git range-diff origin/develop 2d1f3e2657d698de2b3d3a9cdc8947590edc9bbd e4e7c440f48058dabdc571a0a20626b513efd26c
 1:  fb00431b7c =  1:  fb00431b7c Merge bitcoin/bitcoin#22385: refactor: Use DeploymentEnabled to hide VB deployments
 2:  8928146bfa =  2:  8928146bfa Merge bitcoin/bitcoin#22550: test: improve `test_signing_with_{csv,cltv}` subtests (speed, prevent timeout)
 3:  cbd2be8e18 =  3:  cbd2be8e18 Merge bitcoin/bitcoin#22597: consensus/params: simplify ValidDeployment check to avoid gcc warning
 4:  c2b61f062b !  4:  fc25503cbc Merge bitcoin/bitcoin#22632: test: Set regtest.BIP66Height = 102 to speed up tests
    @@ Commit message
     
         Tree-SHA512: 4bbee3c8587d612e74a59fde49b6439c1296f2fc27d3a7cf59a35e920f729fdd581c930290bd04def618f81412236676ddb99b4ceb4d80dfb9fd610b128a04b1
     
    - ## doc/release-notes-22632.md (new) ##
    + ## doc/release-notes-6189.md (new) ##
     @@
     +Tests
     +-----
     +
     +- For the `regtest` network the BIP 66 (DERSIG) activation height was changed
    -+  from 1251 to 102. (#22632)
    ++  from 1251 to 102. (dash#6189)
     
      ## src/chainparams.cpp ##
     @@ src/chainparams.cpp: public:
 5:  94483a8c2f =  5:  71af8816ef Merge bitcoin/bitcoin#22907: test: Avoid intermittent test failure in feature_csv_activation.py
 6:  8401431a82 =  6:  101a863399 Merge bitcoin/bitcoin#16333: test: Set BIP34Height = 2 for regtest
 7:  a546f13ff3 <  -:  ---------- Merge bitcoin/bitcoin#22718: doc: Add missing PR 16333 release note
 -:  ---------- >  7:  adcf095ab9 Merge bitcoin/bitcoin#22718: doc: Add missing PR 16333 release note
 8:  a584b4da0b !  8:  65b92fa093 Merge bitcoin/bitcoin#21862: test: Set regtest.BIP65Height = 111 to speed up tests
    @@ Commit message
     
         Tree-SHA512: 79a8263e7233838666b9b636b496a8b9eb12398c779f9434677e1d62816732c0a7c7b3e73965be1fb0038d35e05e5a90e665bd74e9610104127dfc4ea38169bf
     
    - ## doc/release-notes-22632.md ##
    -@@ doc/release-notes-22632.md: Tests
    -   changed.
    -   * BIP 34 (blockheight in coinbase) from 500 to 2 (#16333)
    -   * BIP 66 (DERSIG) from 1251 to 102 (#22632)
    -+  * BIP 65 (CLTV) from 1351 to 111 (#21862)
    + ## doc/release-notes-6189.md ##
    +@@ doc/release-notes-6189.md: Tests
    +   changed. (dash#6189)
    +   * BIP 34 (blockheight in coinbase) from 500 to 2
    +   * BIP 66 (DERSIG) from 1251 to 102
    ++  * BIP 65 (CLTV) from 1351 to 111
     
      ## src/chainparams.cpp ##
     @@ src/chainparams.cpp: public:
 9:  90e84fd2c8 =  9:  e4e7c440f4 fix: use proper chain instead using ActiveChain for test framework
10:  2d1f3e2657 <  -:  ---------- Update doc/release-notes-22632.md
```
